content/articles/2018-02-devtools-refactor.Rmd

```diff
@@ -0,0 +1,65 @@
+---
+title: The refactoring of devtools
+slug: devtools-refactor
+description: >
+  Refactoring devtools: why and how
+author: Jenny Bryan
+date: 2018-02-26
+categories:
+  - package
+tags: []
+photo:
+  url: https://morguefile.com/p/734816
+  author: mconnors
+---
+
+```{r setup, include = FALSE}
+library(usethis)
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+[devtools](https://github.com/r-lib/devtools#readme) is an R package that helps you create, develop, and install R packages. By 2017, it was clear that devtools would be easier to maintain if it was split into a few smaller and more focused packages. For example, there is internal tension from trying to serve two radically opposing aims: a batteries-included framework for rapid iteration in local package development and a minimal-dependency package installer and checker, e.g., for continuous integration settings like [Travis CI](https://travis-ci.org).
```

Comment:
The shift between the first and second sentences feels a little off to me. I know we're doing the "what is the package" line, but without an anchoring date the _"By..."_ construct is a little awkward.